src/runtime/ndarray.cc

```diff
@@ -78,6 +77,10 @@ struct NDArray::Internal {
   // This enables us to create NDArray from memory allocated by other
   // frameworks that are DLPack compatible
   static void DLPackDeleter(NDArray::Container* ptr) {
+    // if the array is pinned by dgl, unpin it before freeing
+    if (ptr->from_tensor_dispatcher_ && ptr->pinned_by_dgl_)
```

Comment:
`if (... && ptr->IsContainerPinned())`

Reply:
If we borrowed a tensor pinned by PyTorch, we don't want to unpin it automatically.